sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathProperties.cs

```diff
@@ -171,6 +171,41 @@ public class PathProperties
         /// </summary>
         public DateTimeOffset AccessTierChangedOn { get; internal set; }
 
+        /// <summary>
+        /// If this path represents a directory.
+        /// </summary>
+        public bool IsDirectory
+        {
+            get
+            {
+                Metadata.TryGetValue(Constants.DataLake.IsDirectoryKey, out string isDirectory);
+                if (isDirectory != null && isDirectory == bool.TrueString.ToLowerInvariant())
+                {
+                    return true;
+                }
+                return false;
```

Comment:
Would it be safer to use `bool.TryParse`?  You could turn this whole expression into

```C#
get => Metadata.TryGetValue(Constants.DataLake.IsDirectoryKey, out string isDirectoryValue) &&
    bool.TryParse(isDirectoryValue, out bool isDirectory) &&
    isDirectory;
```

Reply:
fixed